### PR TITLE
Accelerate

### PR DESCRIPTION
**Pull request by erogol**
Intorduces HF Accelerate as an alternative training backend. Hopefully it'll ease integration of Deepspeed going for scalable training runs.

### CODE REVIEW

**Review by Edresson — CHANGES_REQUESTED** [review]

File: trainer/trainer.py [hunk 1]
```diff
@@ -553,6 +556,42 @@ def __init__(  # pylint: disable=dangerous-default-value
...
+
+    @property
+    def use_pt_ddp(self):
+        """Return True if using PyTorch DDP."""
+        return self.num_gpus > 1 and not self.args.use_accelerate
+
+    @property
+    def use_accelerate(self):
+        """Return True if using HF Accelerate."""
+        return self.num_gpus > 1 and self.args.use_accelerate
```
[the remark this answers is not in the document]
Comment (erogol) [reply trainer/trainer.py:572]: it can be used for single GPU too. To be able to use some of its perks like deepspeed, TPU training, etc

**Review by WeberJulian — COMMENTED** [review]
I'm running a hifigan training with `TrainerArgs(use_accelerate=True)` to check if it works fine.

File: trainer/trainer.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (erogol) [reply trainer/trainer.py:572]: yeah just encounter it while training. Dropping num_gpu check.

File: trainer/trainer.py [hunk 2]
```diff
@@ -553,6 +556,42 @@ def __init__(  # pylint: disable=dangerous-default-value
         self.save_training_script()
         ping_training_run()
 
+    @property
+    def use_apex(self):
+        """Return True if using APEX."""
+        return self.num_gpus > 1 and not self.args.use_accelerate and self._is_apex_available()
```
[the remark this answers is not in the document]
Comment (erogol) [reply trainer/trainer.py:562]: it shouldnt 👍

**Review by Edresson — APPROVED** [review]
Looks all good to me :)

File: requirements.dev.txt [hunk 3]
```diff
@@ -3,3 +3,4 @@ coverage
 isort
 pytest
 pylint
+accelerate  # for testing
```
[the remark this answers is not in the document]
Comment (erogol) [reply requirements.dev.txt:6]: I don't want to mandate accelerate and it is weird that it is only in dev so I know I'll be confused. It is for my future self :)

**Pull request merged**